src/betterproto/compile/naming.py

```diff
@@ -11,3 +13,11 @@ def pythonize_field_name(name: str) -> str:
 
 def pythonize_method_name(name: str) -> str:
     return casing.safe_snake_case(name)
+
+
+def pythonize_enum_member_name(name: str, enum_name: str) -> str:
+    enum_name = casing.snake_case(enum_name).upper()
+    find = name.find(enum_name)
```

Comment:
I don't think removing the enum name from just anywhere in the key name would be a good idea, especially if each key is evaluated independently.

```suggestion
    prefix = casing.snake_case(enum_name).upper() + "_"
    if name.startswith(prefix):
        return name[len(prefix):]
    return name
```

I wonder if it's better to be all or nothing here... i.e. only do the renaming if the convention is perfectly followed (by all keys). Do you know of a precedent or documented rule to follow one way or the other?

Reply:
I used find because sometimes people in C use k_ to prefix an enum.

Reply:
+1 for find, although a loosely held opinion

Reply:
Why would a prefix of `k_` necessitate `str.find`? Just curious

Do we just need a small test for this case (and any others)?

Reply:
Because the name doesn't start with k_

Reply:
I still think this implementation is unclear and error prone. For example imagine the chaos given an enum type with a single letter name! If the intention is to remove a "prefix", then just remove the prefix.

If you feel it's necessary to make it more clever than that (which I doubt) then a docstring explanation and test coverage are essential.